### PR TITLE
Add Strapi authentication

### LINKED ISSUE #4: Add strapi Authentication to cron_worker
As part of the epic, we need to add strapi authentication to the cron-worker.

### PR DESCRIPTION
**Pull request by dahuang37**
adds a new file `strapi-api.ts` for implementing API calls to Strapi.
It encapsulate the auth implementation in its class and provide a few
example function for calling to strapi endpoint.

Bonus:
- refactored the auth to google-api, which also encapsulate the auth
functionality in the file and only expose the API functionality.
- refactored handler to use the new google-api and strapi-api.
- Fixes the issue with parsing admin reports from Google API. 

Testing:
works locally with miniflare. additional testing is required for
cloudflare worker.

Resolves #4 

Signed-off-by: Daniel Huang <danielhuang37@outlook.com>

### CODE REVIEW

File: src/strapi-api.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if (!username || !password) {
+            username = this.env.STRAPIUSERNAME;
+            password =  this.env.STRAPIPASSWORD;
+        }
+
+        let data = {
+            "identifier": username,
+            "password": password
+        }
+        let strapiResponse = await fetch("http://localhost:1337/auth/local", {
```
Comment (chungthuang) [inline src/strapi-api.ts]: Should we change this to `STRAPIBASEURL`?

File: src/strapi-api.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private async fetch(urlPath: string) {
+        if (!this.token) {
+            this.token = await this.getToken()
+        }
+
+        let url = this.env.STRAPIBASEURL + urlPath
+        const response = await fetch(url, {
+            method: 'GET',
+            headers: {
+            'cookie': this.token
```
Comment (dahuang37) [inline src/strapi-api.ts]: indent

File: src/strapi-api.ts (hunk 1 above)
Comment (dahuang37) [reply src/strapi-api.ts]: updated.

File: src/strapi-api.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        })
+        const result = await response.json<RespJson>();
+        return result
+    }
+
+    private async getToken(): Promise<string> {
+
+        var username: string = await cron_worker.get("strapi_username") as string;
+        var password: string = await cron_worker.get("strapi_password") as string;
+        if (!username || !password) {
```
Comment (chungthuang) [inline src/strapi-api.ts:39]: Can we comment that this is for dev environment?

**Review by chungthuang — APPROVED** [review]

File: src/strapi-api.ts (hunk 3 above)
Comment (dahuang37) [reply src/strapi-api.ts:39]: added comment

File: src/google-api.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import * as jose from 'jose'
+import type { RespJson } from "./common"
+
+declare global {
+    const cron_worker: KVNamespace;
+}
+
+export class GoogleAPI {
+    authenticator: GoogleJwtAuthenticater;
+    adminScopes = "https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/admin.directory.group https://www.googleapis.com/auth/admin.directory.group.member https://www.googleapis.com/auth/admin.reports.audit.readonly https://www.googleapis.com/auth/admin.directory.user.security" ;
```
Comment (chungthuang) [inline src/google-api.ts:11]: Can this be an array? It's easier to read that way.

Comment (dahuang37) [reply src/google-api.ts:11]: the scope has to be a single string and not an array. Added comments to explain the issue.

**Pull request merged**